### PR TITLE
fix: ethereum proxy

### PR DESCRIPTION
**Pull request by leoslr**
## Description of the changes

Deployment of the latest version of EthereumProxy and EthereumFeeProxy which enables payments to/from smart wallet.
Deployment made through the RequestDeployer with`CREATE2`.

Update versions in advanced logic & payment detection

### CODE REVIEW

File: packages/request-client.js/test/etherscan-mock.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export default class EtherscanProviderMock {
           to: '0xc12F17Da12cd01a9CDBB216949BA0b41A6Ffc4EB',
           value: BigNumber.from('0x02dd231b00'),
           nonce: 205,
-          data: '0xc19da4923539c37f',
+          data: '0xefce79375b2db9f7',
           creates: null,
```
Comment (leoslr) [inline packages/request-client.js/test/etherscan-mock.ts:44]: As we use a new contract version, the requestId is modified resulting in change of payment reference.

**Review by bertux — APPROVED** [review]

**Review by KolevDarko — COMMENTED** [review]
These all look good to me, I mean they're just mostly configs, but if they pass the tests it should be good.
However I think the final review before using these in production should be from @yomarion

File: packages/request-client.js/test/index.test.ts [hunk 2]
```diff
@@ -1630,25 +1630,19 @@ describe('index', () => {
...
       expect(dataAfterRefresh.balance?.events[0].amount).toBe('12300000000');
       expect(dataAfterRefresh.balance?.events[0].parameters!.txHash).toBe(
         '0x06d95c3889dcd974106e82fa27358549d9392d6fee6ea14fe1acedadc1013114',
       );
-
-      expect(dataAfterRefresh.balance?.events[1].name).toBe('payment');
-      expect(dataAfterRefresh.balance?.events[1].amount).toBe('45600000');
-      expect(dataAfterRefresh.balance?.events[1].parameters!.txHash).toBe(
-        '0x38c44820c37d31fbfe3fcee9d4bcf1b887d3f90fb67d62d924af03b065a80ced',
-      );
```
[the remark this answers is not in the document]
Comment (benjlevesque) [reply packages/request-client.js/test/index.test.ts:1651]: maybe it should be replaced by a transaction on rinkeby? or be left to use the previous version?

**Review by benjlevesque — APPROVED** [review]

**Review by yomarion — APPROVED** [review]

File: packages/request-client.js/test/index.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (leoslr) [reply packages/request-client.js/test/index.test.ts:1651]: I am not convinced that's a good idea to rely on existing transaction for testing purposes. 
Also, I wanted to keed the previous version and add a new test, but when we create the request in the test it automatically uses the latest proxy version, which I believe I cannot choose manually.

**Pull request merged**